x-pack/libbeat/management/fleet/manager.go

```diff
@@ -122,38 +126,52 @@ func (cm *Manager) CheckRawConfig(cfg *common.Config) error {
 	return nil
 }
 
+// UpdateStatus updates the manager with the current status for the beat.
+func (cm *Manager) UpdateStatus(status management.Status, msg string) {
+	cm.lock.Lock()
+	if cm.status != status || cm.msg != msg {
+		cm.status = status
+		cm.msg = msg
+		cm.lock.Unlock()
+		cm.client.Status(statusToProtoStatus(status), msg)
+		cm.logger.Infof("Status change to %s: %s", status, msg)
+		return
+	}
+	cm.lock.Unlock()
+}
+
 func (cm *Manager) OnConfig(s string) {
-	cm.client.Status(proto.StateObserved_CONFIGURING, "Updating configuration")
+	cm.UpdateStatus(management.Configuring, "Updating configuration")
```

Comment:
should we switch back after config is applied or do we let beat to do that

Reply:
It does switch back to Running at the end of `OnConfig`.

Reply:
you;re right it was hidden so i overlooked

Reply:
If I read this right OnConfig sets the status to 'Healthy' at the end. Are we sure this is the correct state to report? Do we 'forget' the internal state here?

How about removing the 'Configuring' status? The less states (and protocol behavior) we have,  the less we can get it wrong.

Reply:
I personally do not like the idea of setting the status in the `OnConfig` handler, I feel like that should really only be done specifically by the beat itself. So if OnConfig passes it to the reloader then the reloader should really set the `Status`.

At the moment I keep it this way so it works, and once status is being reported correctly by the beat, this should be removed (probably at the same time).

I do not think we should removing `Configuring` I think it is a key state for watching the observability of the Beat and propogating that information to Fleet.

It is good to know that going `Healthy -> Configuring -> Failed` is because everything was good, then updating the configuration caused it to fail.

Reply:
> I personally do not like the idea of setting the status in the OnConfig handler, I feel like that should really only be done specifically by the beat itself. So if OnConfig passes it to the reloader then the reloader should really set the Status.

> At the moment I keep it this way so it works, and once status is being reported correctly by the beat, this should be removed (probably at the same time).

+1 
Is this required change/cleanup documented in an issue?